core/src/main/java/com/linecorp/armeria/common/stream/ByteBufDeframerInput.java

```diff
@@ -0,0 +1,216 @@
+                    value |= buf.readUnsignedByte();
+                    break;
+                case 2:
+                    value |= buf.readUnsignedShort();
+                    break;
+                case 3:
+                    value |= buf.readUnsignedMedium();
+                    break;
+                default:
+                    throw new Error(); // Should not reach here.
+            }
+            remaining -= readSize;
+            if (remaining == 0) {
+                break;
+            }
+        }
+
+        if (remaining == 0) {
+            return value;
+        } else {
+            throw newEndOfInputException();
+        }
+    }
+
+    @Override
+    public ByteBuf readBytes(int length) {
+        checkArgument(length > 0, "length %s (expected: length > 0)", length);
+        final ByteBuf firstBuf = queue.peek();
+        if (firstBuf == null) {
+            throw newEndOfInputException();
+        }
+
+        final int readableBytes = firstBuf.readableBytes();
+        if (readableBytes == length) {
+            return queue.remove();
+        }
+        if (readableBytes > length) {
+            return firstBuf.readRetainedSlice(length);
+        }
+
+        return readBytesSlow(length);
+    }
+
+    private ByteBuf readBytesSlow(int length) {
+        ByteBuf value = null;
+        int remaining = length;
+        for (final Iterator<ByteBuf> it = queue.iterator(); it.hasNext();) {
+            final ByteBuf buf = it.next();
+            if (!buf.isReadable()) {
+                it.remove();
+                buf.release();
+                continue;
+            }
+
+            final int readableBytes = buf.readableBytes();
+            if (value == null) {
+                if (readableBytes == length) {
+                    it.remove();
+                    return buf;
+                }
+
+                if (readableBytes > length) {
+                    return buf.readRetainedSlice(length);
+                }
+
+                value = alloc.buffer(length);
+            }
+
+            final int readSize = Math.min(remaining, readableBytes);
+            value.writeBytes(buf, readSize);
+            if (readableBytes == readSize) {
+                it.remove();
+                buf.release();
+            }
+
+            remaining -= readSize;
+            if (remaining == 0) {
+                break;
+            }
+        }
+
+        if (remaining > 0 || value == null) {
+            ReferenceCountUtil.release(value);
+            throw newEndOfInputException();
+        }
+
+        return value;
+    }
+
+    /**
+     * Discards the {@link ByteBuf}s that have been fully consumed and are not readable anymore.
+     */
+    void discardReadBytes() {
+        for (;;) {
+            final ByteBuf buf = queue.peek();
+            if (buf != null && !buf.isReadable()) {
+                queue.remove().release();
+            } else {
+                break;
+            }
+        }
+    }
```

Comment:
Is this still needed?

Reply:
`discardReadBytes()` was removed after fixing readInt() and readByte() together.